hera_cal/redcal.py

```diff
@@ -286,6 +286,7 @@ def logcal(self, data, sol0={}, wgts={}, sparse=False):
             sol: dictionary of gain and visibility solutions in the {(index,antpol): np.array}
                 and {(ind1,ind2,pol): np.array} formats respectively
         """
+        import linsolve
```

Comment:
To truly make this optional without breaking tests, we need something with the following syntax (borrowed from pyuvdata):
```
try:
    import linsolve
except(ImportError):
    from nose.plugins.skip import SkipTest
    raise SkipTest('linsolve not detected. linsolve'
                   ' must be installed for this functionality')
```
This should go in each function that wants to import linsolve